### PR TITLE
Subprocess stop fixes

### PR DESCRIPTION
**Pull request by bbaldino**
this adds some better handling for a couple exception scenarios around stopping a subprocess

### CODE REVIEW

**Comment by bbaldino** [comment]
Want to double check the pid change on an actual jibri machine, but it worked fine on my mac.

**Comment by bbaldino** [comment]
tested the pid change on a jibri machine, looks good.

File: src/test/kotlin/org/jitsi/jibri/util/ProcessWrapperTest.kt [hunk 1]
```diff
@@ -161,5 +163,29 @@ internal class ProcessWrapperTest : ShouldSpec() {
...
+                }
+
+            }
+            "when the runtime throws RuntimeException" {
+                whenever(runtime.exec(anyString())).thenAnswer { throw RuntimeException() }
+                should("handle it correctly") {
+                    processWrapper.stopAndWaitFor(Duration.ofSeconds(10)) shouldBe false
+                }
+            }
+        }
```
Comment (virtuacoplenny) [inline src/test/kotlin/org/jitsi/jibri/util/ProcessWrapperTest.kt]: destroyForciblyAndWaitFor tests?

File: src/main/kotlin/org/jitsi/jibri/util/JibriSubprocess.kt [hunk 2]
```diff
@@ -55,31 +56,32 @@ class JibriSubprocess(
...
+
+    fun stop() {
+        logger.info("Stopping $name process")
+        processStatePublisher?.stop()
+
+        process?.apply {
+            if (!stopAndWaitFor(Duration.ofSeconds(10))) {
+                logger.error("Error trying to gracefully stop $name, destroying forcibly")
+                if (!destroyForciblyAndWaitFor(Duration.ofSeconds(10))) {
+                    logger.error("Error trying to destroy $name forcibly")
```
Comment (virtuacoplenny) [inline src/main/kotlin/org/jitsi/jibri/util/JibriSubprocess.kt]: Follow up PR maybe to handle if forced stop failed.

File: src/test/kotlin/org/jitsi/jibri/util/ProcessWrapperTest.kt (hunk 1 above)
Comment (bbaldino) [reply src/test/kotlin/org/jitsi/jibri/util/ProcessWrapperTest.kt]: fixed

File: src/main/kotlin/org/jitsi/jibri/util/JibriSubprocess.kt (hunk 2 above)
Comment (bbaldino) [reply src/main/kotlin/org/jitsi/jibri/util/JibriSubprocess.kt]: yeah i'm thinking we should go to an error state here.  will do that in another PR

**Pull request merged**